backend/dataall/modules/shares_base/services/share_item_service.py

```diff
@@ -204,3 +216,53 @@ def list_shareable_objects(share, filter, is_revokable=False):
     def paginated_shared_with_environment_datasets(session, uri, data) -> dict:
         share_item_shared_states = ShareStatusRepository.get_share_item_shared_states()
         return ShareObjectRepository.paginate_shared_datasets(session, uri, data, share_item_shared_states)
+
+    @staticmethod
+    @ResourcePolicyService.has_resource_permission(APPROVE_SHARE_OBJECT, parent_resource=_get_share_uri)
+    def update_filters_table_share_item(uri: str, data: dict):
+        context = get_context()
+        with context.db_engine.scoped_session() as session:
+            if share_item := ShareObjectRepository.get_share_item_by_uri(session, uri):
+                if share_item.itemType != ShareableType.Table.value:
+                    raise Exception(f'Share item is not type {ShareableType.Table.value} - required for data filters')
+
+                if share_item.status in ShareStatusRepository.get_share_item_shared_states():
+                    raise Exception(f'Share item already shared in state {share_item.status} - can not assign filters')
+                try:
+                    if share_item.attachedDataFilterUri:
+                        share_item_filter = ShareObjectItemRepository.get_share_item_filter_by_uri(
+                            session, share_item.attachedDataFilterUri
+                        )
+                        ShareObjectItemRepository.update_share_item_filters(session, share_item_filter, data)
+                        return True
+
+                    share_item_filter = ShareObjectItemRepository.create_share_item_filters(session, share_item, data)
+                    share_item.attachedDataFilterUri = share_item_filter.attachedDataFilterUri
+                    return True
+                except exc.IntegrityError:
+                    raise InvalidInput(
+                        'label',
+                        data.get('label'),
+                        f'same label already exists on another share item for table {share_item.itemName}',
+                    )
+            raise ObjectNotFound('ShareObjectItem', uri)
+
+    @staticmethod
+    def get_share_item_data_filters(uri: str):
+        with get_context().db_engine.scoped_session() as session:
+            return ShareObjectItemRepository.get_share_item_filter_by_uri(session, uri)
+
+    @staticmethod
+    @ResourcePolicyService.has_resource_permission(
+        APPROVE_SHARE_OBJECT, parent_resource=_get_share_uri_from_item_filter_uri
+    )
+    def remove_share_item_data_filters(uri: str):
+        with get_context().db_engine.scoped_session() as session:
+            share_item = ShareObjectItemRepository.get_share_item_by_item_filter_uri(session, uri)
+            share_item.attachedDataFilterUri = None
```

Comment:
I am not 100% sure, but will this line update the attachedDataFilters even when the exception in line 263 is raised?

Reply:
Hmmm - good catch I think it might, moving `share_item.attachedDataFilterUri = None` to after the check / raise of Exception

Reply:
resolved above

Reply:
checked, all good!